src/Avalonia.Native/WindowImpl.cs

```diff
@@ -103,6 +103,15 @@ public WindowState WindowState
 
         public Thickness OffScreenMargin { get; } = new Thickness();
 
+        public IntPtr? ZOrder
+        {
+            get
+            {
+                // macOS returns z-order in reverse order - the topmost window has the lowest z-order
+                return new IntPtr(-_native.WindowZOrder.ToInt64());
```

Comment:
nit: this negation should happen on native side, since technically Avalonia.Native project isn't supposed to be macOS specific.

Reply:
Alright, changed